### PR TITLE
Move entire controller and authenticate step over from app

### PR DESCRIPTION
**Pull request by jkempster34**
Because the wizard will likely only used by the GiT services, move over some extra code to dry up the apps:
- the rest of the controller code
- authenticate step

Copy over the dummy app from a new Rails engine to test the controller concern against

### CODE REVIEW

**Review by ethax-ross — COMMENTED** [review]
Looks good, but should there be some tests to cover this? I think that would also highlight that the API client isn't a dependency in the gemspec

**Comment by jkempster34** [comment]
> Looks good, but should there be some tests to cover this? I think that would also highlight that the API client isn't a dependency in the gemspec

Not sure how to test this without adding Rails to the Gem creating a fake controller 

```
class FakeController < ApplicationController
    include DFEWizard::Controller
end
```

File: spec/dummy/app/views/event_steps/show.html.erb [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<% @hide_page_helpful_question = true %>
```
Comment (ethax-ross) [inline spec/dummy/app/views/event_steps/show.html.erb:1]: Think this is redundant here and in the update template?

**Review by ethax-ross — COMMENTED** [review]
I think this looks good, but its hard to tell if its all tests as its such a big diff; is it worth pulling in simplecov to make sure there aren't any paths missed out?

**Pull request merged**